### PR TITLE
add RemainderModel for general purpose splitting

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

**Comment by simonbyrne** [comment]
TODO: need to benchmark.

**Comment by mwarusz** [comment]
I ran some benchmarks on the rising thermal bubble and it seems that `RemainderModel` is as fast as `AtmosAcousticNonlinearModel` so the latter can be removed :thumbsup:

File: src/Utilities/VariableTemplates/VariableTemplates.jl [hunk 1]
```diff
@@ -143,6 +144,7 @@ Grad{S}(array) where {S} = Grad{S,typeof(array),0}(array)
 
 Base.eltype(g::Grad) = eltype(getfield(g,:array))
 Base.propertynames(::Grad{S}) where {S} = fieldnames(S)
+Base.similar(g::Grad{S,A,offset}) where {S,A,offset} = Grad{S,A,offset}(similar(getfield(g,:array)))
```
Comment (charleskawczynski) [inline src/Utilities/VariableTemplates/VariableTemplates.jl]: Is this tested in the tests? Maybe good to add.

Comment (simonbyrne) [reply src/Utilities/VariableTemplates/VariableTemplates.jl]: Good point.

**Pull request merged**